homeassistant/components/sensor/dominionenergy.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Washington Dominion Energy.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dominion_energy/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+HOURS_TO_UPDATE = timedelta(hours=24)
+CURRENT_BILL_SELECTOR = str("#homepageContent > div:nth-child(3) >"
+                            " div:nth-child(2) > p > span")
+REQUIREMENTS = ['selenium==3.11.0']
+
+DEFAULT_NAME = "Dominion Energy"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and getting the username and password."""
+
+    sensor = DominionEnergySensor(config[CONF_NAME], config[CONF_USERNAME],
+                                  config[CONF_PASSWORD])
+    sensor.update()
+    if sensor.valid_credentials is True:
+        add_devices([sensor])
+    else:
+        _LOGGER.error("Setup DominionEnergy Fail"
+                      " check if your username/password changed")
+        return
+    return
+
+
+class DominionEnergySensor(Entity):
+    """Washington Dominion Energy Sensor will check the bill on daily basis."""
+
+    def __init__(self, name, username, password):
+        """Initialize the sensor."""
+        self._state = None
+        self._password = password
+        self._username = username
+        self._name = name
+        self.valid_credentials = False
```

Comment:
Is this variable really necessary? I mean every time the credentials are invalid `self._state` will be `None`.